### ISSUE TITLE
NetRender doesn't connect with localhost

### ISSUE THREAD

**Issue by buddhi1980** [buddhi1980/mandelbulber2#258]
##### System information (version)
-  mandelbulber2 => 2.10
- Operating System / Platform => All
- Compiler => gcc, mingw

##### Detailed description
When client is on the same machine as server then there is generated connection error

```
Warning: QAbstractSocket::connectToHost() called when already looking up or connecting/connected to "localhost" (:0
```

Similar error was reported here: http://www.fractalforums.com/bug-reporting-b231/version-2-10-clients-crashing/new/?topicseen#new

##### Steps to reproduce

start NetRender server and client on the same machine and try connect them together.

**Comment by zebastian** [comment]
i found a problem, which probably caused this behaviour:

The client tries to reconnect to the server, as long as it is not in the state QAbstractSocket::ConnectedState.
see: void CNetRender::TryServerConnect()

If the client is already trying to connect to the server, it should do nothing (since the establishing of the connection may take longer than 1 second). If it is not already connecting, it should reset the socket and start a connection, which effectively should only be run once at the beginning, or while the server is unreachable.

The problem is: client recognized only the "ConnectingState" as the pending state, but there is also a HostLookupState which also resolves to an "establishing connection state"
If this takes too long (even on localhost i could reprodouce this) the socket gets resetted every second before it can finish the lookup, hence the error log.

Adding HostLookupState as a "wait for result" state seems to fix the problem, see commit: https://github.com/buddhi1980/mandelbulber2/commit/b8eb297e4ec68a56b32e6d2e4a0d92b5e16ec49a

**Comment by buddhi1980** [comment]
Works perfect! Thanks!

**Comment by CCV54** [comment]
Yep, thanks zebastian! Fixed my problem, as mentioned by buddhi1980 - http://www.fractalforums.com/bug-reporting-b231/version-2-10-clients-crashing/ - and a little more.

**Comment by mclarekin** [comment]
@ zebastian. Note the Gamepad comment and another minor bug comment from
CCV's post today.













*About the slow loading on Client end:Turns out, it was my fault due to
copying the already extracted mandelbulber2-win64-2.10 file from my Server
machine to Clients.After I copied the Zip file instead and extracted the
contents in place, the (newly created) Desktop shortcut works
fine.Excellent work, people, and thank you all very much.  [image: A Beer
Cup] [image: A Beer Cup] [image: A Beer Cup]The thing about Gamepad:I
wonder.. Does it need to be enabled, by default, at launch? Might there be
some way to choose whether it is or it isn't?And, another minor bug:Clients
are often displayed as READY - on both ends - and I find it takes some
various fiddling to get them to show as WORKING.It happens that, the few
times I checked and despite what the display says, the client is actually
working. [image: huh?]*
